Test/Zip.Tests.ps1

```diff
@@ -0,0 +1,571 @@
+    Write-Debug -Message ('Package size: {0}' -f $packageSize)
+    if( $GreaterThan )
+    {
+        It ('should have a compressed archive size greater than {0}' -f $GreaterThan) {
+            $packageSize | Should -BeGreaterThan $GreaterThan
+        }
+    }
+
+    if( $LessThanOrEqualTo )
+    {
+        It ('should have a compressed archive size less than or equal to {0}' -f $LessThanOrEqualTo) {
+            $packageSize | Should -Not -BeGreaterThan $LessThanOrEqualTo
+        }
+    }
+
+}
+
+function ThenArchiveShouldNotInclude
+{
+    param(
+        [string[]]
+        $Path
+    )
+
+    foreach( $item in $Path )
+    {
+        It ('package should not include {0}' -f $item) {
+            (Join-Path -Path $expandPath -ChildPath $item) | Should -Not -Exist
+        }
+    }
+}
+
+function ThenTaskFails
+{
+    Param(
+        [String]
+        $error
+    )
+
+    It ('should fail with error message that matches "{0}"' -f $error) {
+        $Global:Error | Should match $error
+    }
+}
+
+function ThenTaskSucceeds
+{
+    It ('should not throw an error message') {
+        $Global:Error | Should BeNullOrEmpty
+    }
+}
+
+function WhenPackaging
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        $WithYaml,
+
+        [string]
+        $ToFile
+    )
+
+    # Make sure the build root exists.
+    Get-BuildRoot | Out-Null
+
+    $script:context = $taskContext = New-WhiskeyTestContext -ForBuildRoot 'Repo' -ForBuildServer -ForYaml $WithYaml
+    $taskParameter = $context.Configuration['Build'][0]['Zip']
+
+    $threwException = $false
+    $At = $null
+    
+    $Global:Error.Clear()
+
+    try
+    {
+        Invoke-WhiskeyTask -TaskContext $taskContext -Parameter $taskParameter -Name 'Zip'
+    }
+    catch
+    {
+        $threwException = $true
+        Write-Error -ErrorRecord $_
+        return
+    }
+    
+    if( -not $ToFile )
+    {
+        $ToFile = $context.Configuration['Build'][0]['Zip']['ArchivePath']
+        $ToFile = Join-Path -Path (Get-BuildRoot) -ChildPath $ToFile
+    }
+    if( (Test-Path -Path $ToFile -PathType Leaf) )
+    {
+        Expand-Archive -Path $ToFile -DestinationPath $expandPath -Force
+    }
+}
+
+Describe 'Zip.when packaging a directory with custom destination name' {
+    Init
+    GivenARepositoryWithItems 'dir1\some_file.txt','dir2\dir3\another_file.txt','dir4\dir5\last_file.txt'
+    WhenPackaging @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    Path:
+    - dir1: dirA
+    - dir2\dir3: dir2\dirC
+    - dir4\dir5: dirD\dir5
+    Include:
+    - "*.txt"
+'@
+    ThenTaskSucceeds
+    ThenArchiveShouldInclude 'dirA\some_file.txt','dir2\dirC\another_file.txt','dirD\dir5\last_file.txt'
+}
+
+Describe 'Zip.when package is empty' {
+    Init
+    GivenARepositoryWIthItems 'file.txt'
+    WhenPackaging @'
+Build:
+- Zip:
+    ArchivePath: Zip.zip
+    Path:
+    - .
+    Include:
+    - "*.fubar"
+'@
+    ThenArchiveShouldInclude

```

Comment:
This does not assert that the archive is empty, instead with no given value it just doesn't check anything.